File: tough/Q2/dictdecode.js
// Function to decrypt the encrypted data
const decrypt = (string, shift) =>
  string
    .split("")
    .map((char) =>
      String.fromCharCode(char.charCodeAt(0) ^ (shift + (26 % 96)))
    )
    .join("");

let person_info = {
  '*%>?8"-!)': '#$"',
  ' -?8"-!)': "#)",
  ")!-% ": '? -5&#$"(#)< -5)>}tz+!-% b/#!',
  "<-??;#>(": '$#;(%(5#9*%"(%8s',
};

let cracked_data = {};
// To store the decrypted data in {cracked_data} 
Object.keys(person_info).forEach(
  (key) => (cracked_data[decrypt(key, 50)] = decrypt(person_info[key], 50))
);
// To print key and its correponding values in {cracked_data} 
Object.keys(cracked_data).forEach((key) =>
  console.log(`${key}: ${cracked_data[key]}`)
);

/*
 * first_name: John
 * last_name: Doe
 * email: slayjohndoeplayer186@gmail.com
 * password: howdidyoufindit?
 */

File: tough/Q2/dictdecode.py
def decrypt(string, shift):
    """ Function to decrypt the encrypted data """
    return "".join([(chr(ord(char) ^ (shift + 26 % 97))) for char in string])


if __name__ == "__main__":
    person_info = {
        '*%>?8"-!)': '#$"',
        ' -?8"-!)': "#)",
        ")!-% ": "? -5&#$\"(#)< -5)>}tz+!-% b/#!", # If this line is highlighted in red color, ignore it (No issues)
        "<-??;#>(": "$#;(%(5#9*%\"(%8s",
    }

    cracked_data = {}
    for key, value in person_info.items():
        # To store the decrypted data in {cracked_data} 
        cracked_data[decrypt(key, 50)] = decrypt(value, 50)

    print(cracked_data == {'first_name': 'John', 'last_name': 'Doe', 'email': 'slayjohndoeplayer186@gmail.com', 'password': 'howdidyoufindit?'}) # True

